azure-pipelines.yml

```diff
@@ -65,4 +65,69 @@ steps:
     workingDirectory: '$(modulePath)'
     displayName: 'diagnostics'
 
-  - template: ci/publish-docker-image-steps.yml
+  - task: Docker@1
+    displayName: 'Docker-hub registry login'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    inputs:
+      command: login
+      containerregistrytype: Container Registry
+      dockerRegistryEndpoint: dockerhub
+
+  - task: Docker@1
+    displayName: 'Build image'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    inputs:
+      command: build
+      dockerFile: $(modulePath)/cnab/Dockerfile
+      imageName: $(Build.Repository.Name):$(Build.BuildNumber)
+      includeLatestTag: true
+      containerregistrytype: Container Registry
+      dockerRegistryEndpoint: dockerhub
+
+  - task: Docker@1
+    displayName: 'Push image'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    inputs:
+      command: push
+      imageName: $(Build.Repository.Name):$(Build.BuildNumber)
+      containerregistrytype: Container Registry
+      dockerRegistryEndpoint: dockerhub
+
+  - task: Docker@1
+    displayName: 'tag edge image'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    inputs:
+      command: tag
+      imageName: $(Build.Repository.Name):$(Build.BuildNumber)
+      arguments: $(Build.Repository.Name):edge
+
+  - task: Docker@1
+    displayName: 'Push edge tag'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    inputs:
+      command: push
+      imageName: $(Build.Repository.Name):edge
+      containerregistrytype: Container Registry
+      dockerRegistryEndpoint: dockerhub
+
+  - script: |
+      echo "##vso[task.setvariable variable=version]`cat VERSION`"
+    displayName: 'set version variable for tagging images'
+    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/heads/v'))
```

Comment:
Now also checking for branch name ending with `x`. 

> Related: Let's avoid the complexity of release branches until we need them.

This work is in anticipation of cutting a release so that users can pull tag based on a release.